engine/src/test/java/com/redhat/cloud/notifications/processors/email/aggregators/AdvisorEmailAggregatorTest.java

```diff
@@ -175,4 +185,206 @@ void testAggregate() {
+            );
+        }
+
+        // Shuffle the collection so that we end up with a random order for the
+        // collection of rules.
+        Collections.shuffle(testRuleCollection);
+
+        for (final Map<String, String> testRule : testRuleCollection) {
+            advisorEmailAggregator.aggregate(
+                AdvisorTestHelpers.createEmailAggregation(RESOLVED_RECOMMENDATION, testRule)
+            );
+        }
+
+        // Shuffle the collection so that we end up with a random order for the
+        // collection of rules.
+        Collections.shuffle(testRuleCollection);
+
+        for (final Map<String, String> testRule : testRuleCollection) {
+            advisorEmailAggregator.aggregate(
+                AdvisorTestHelpers.createEmailAggregation(DEACTIVATED_RECOMMENDATION, testRule)
+            );
+        }
+
+        // Grab the resulting JSON.
+        final JsonObject resultingAdvisor = JsonObject.mapFrom(advisorEmailAggregator.getContext()).getJsonObject(ADVISOR_KEY);
+
+        final Map<String, Map<String, Object>> newRecommendations = resultingAdvisor.getJsonObject(NEW_RECOMMENDATIONS).mapTo(Map.class);
+        final Map<String, Map<String, Object>> resolvedRecommendations = resultingAdvisor.getJsonObject(RESOLVED_RECOMMENDATIONS).mapTo(Map.class);
+        final Map<String, Map<String, Object>> deactivatedRecommendations = resultingAdvisor.getJsonObject(DEACTIVATED_RECOMMENDATIONS).mapTo(Map.class);
+
+        // The number of rules returned should be the maximum set in the
+        // advisor's class.
+        Assertions.assertEquals(AdvisorEmailAggregator.MAXIMUM_NUMBER_RETURNED_EVENTS, newRecommendations.size(), "unexpected number of resulting rules for the new recommendations");
+        Assertions.assertEquals(AdvisorEmailAggregator.MAXIMUM_NUMBER_RETURNED_EVENTS, resolvedRecommendations.size(), "unexpected number of resulting rules for the resolved recommendations");
+        Assertions.assertEquals(AdvisorEmailAggregator.MAXIMUM_NUMBER_RETURNED_EVENTS, deactivatedRecommendations.size(), "unexpected number of resulting rules for the deactivated recommendations");
+
+        // Add the test rules in the expected order that should be received,
+        // sorted by total risk in descending order.
+        final List<Map<String, String>> testRules = new ArrayList<>(AdvisorEmailAggregator.MAXIMUM_NUMBER_RETURNED_EVENTS);
+        testRules.add(testRule4);
+        testRules.add(testRule2);
+        testRules.add(testRule1);
+        testRules.add(testRule3);
+        testRules.add(testRule5);
+
+        // Test that the new recommendations are equivalent to the test rules
+        // and that they are in the correct order.
+        final List<Map.Entry<String, Map<String, Object>>> newRecommendationsCollection = new ArrayList<>(newRecommendations.entrySet());
+        this.assertRulesAreEqualAndCorrectOrder(testRules, newRecommendationsCollection);
```

Comment:
This works because `testRule` 1 to 5 all have a different risk value. If two rules have the same risk value, their order will be unpredictable. Similarly, if there are 6 rules with the same highest risk value, we can't predict which one will be excluded from the aggregation result. I'm not sure this is a problem though. The current test looks good enough to me.